src/javascript/ContentEditor/EditPanel/HeaderBadges/ReadOnlyBadge.jsx

```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {Chip, Visibility} from '@jahia/moonstone';
+import {useTranslation} from 'react-i18next';
+import styles from './ReadOnlyBadge.scss';
+import {useContentEditorContext, useContentEditorSectionContext} from '~/contexts';
+
+export const ReadOnlyBadge = ({isGlobal, readOnly}) => {
```

Comment:
To simplify, I feel like we can get rid of the props here and do the check when we're displaying ReadOnlyBadge instead.

Reply:
I thought about this, if I delegate isGlobal and the checks on the node and section to a parent component I end up complicating the parent component. Because this badge is also used at field level I would have to do those duplicate checks at field level as well. So I'm not sure it would simplify things, I mean for the badge component itself yes, but everything else badge interacts with would be a bit more complex.